client/blocks/promo-card-block/index.jsx

```diff
@@ -40,7 +41,7 @@ const PromoCardBlock = ( {
 	return (
 		<>
 			<QueryJetpackPlugins siteIds={ [ selectedSiteId ] } />
-			{ jetpackNonAtomic || selectedPlugin || isFetching ? (
+			{ ( ! showOnJetpackNonAtomic && jetpackNonAtomic ) || selectedPlugin || isFetching ? (
```

Comment:
I wonder if it would make more sense to move this business logic outside the component. ie: The caller should determine if the card is relevant and the PromoCardBlock should only be concerned with displaying itself.

In the Stats code, we already do the necessary checks.

Reply:
Sounds good @a8ck3n, [moved its business logic here](https://github.com/Automattic/wp-calypso/pull/71286/commits/e3db9fae1b025865127316bf3da5394325b8a06d)